crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -731,15 +714,15 @@ impl Entities {
 #[repr(C)]
 struct EntityMeta {
     /// The current generation of the [`Entity`].
-    pub generation: u32,
+    pub generation: Option<NonZeroU32>,
```

Comment:
I like the increase in code clarity here -- this makes it harder to forget to handle the case where an entity is out of room for more generations.

Reply:
Honestly this became a big selling point to be with the generation approach, since currently debug will panic and release will just silently overflow and to rectify that discrepancy we're going to need a slowdown in that area regardless... or just assume nobody will ever exhaust generations or hold a stale reference (which honestly might not be unreasonable)